tests/containers/multi_runtime.pm

```diff
@@ -0,0 +1,174 @@
+# SUSE's openQA tests
+#
+# Copyright 2024-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Validate docker and podman network interoperability
+# - validate if podman and docker networking works together
+# - check if firewalld doesn't break either
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base qw(containers::basetest);
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use utils;
+use containers::common qw(install_packages);
+use Utils::Logging 'save_and_upload_log';
+use version_utils qw(is_tumbleweed);
+
+my $port = 8000;
+
+sub run_tests {
+    my $ip_version = shift;
+
+    my $test_image = "registry.opensuse.org/opensuse/nginx";
+    my $test_url = "https://registry.opensuse.org/v2/";
+    my $volumes = '-v $HOME/nginx/nginx.conf:/etc/nginx/nginx.conf:ro,z -v $HOME/nginx:/usr/share/nginx/html:ro,z';
+    my $curl_opts = "-$ip_version -svL -o /dev/null";
+    my @containers;
+
+    # Create rootfull & rootless containers on both podman & docker
+    for my $runtime ("docker", "podman") {
+        for my $type ("root", "rootless") {
+            my $name = "test_${type}_${runtime}_${ip_version}";
+            my $sudo = ($type eq "root") ? "sudo" : "";
+            my $network = "";
+
+            if ($ip_version == 6 && $runtime eq "podman") {
+                # For IPv6 we need to create a network. IPv4 uses NAT. IPv6 uses routing tables
+                $network = "ip6net";
+                assert_script_run "$sudo $runtime network create --ipv6 $network";
+                assert_script_run "$sudo $runtime network inspect $network";
+                $network = "--network $network";
+            }
+
+            script_retry("$sudo $runtime pull $test_image", retry => 3, delay => 60, timeout => 180);
+            assert_script_run "$sudo $runtime run -d --name $name $network -p $port:80 $volumes $test_image";
+            push @containers, {name => $name, runtime => $runtime, sudo => $sudo, port => $port};
+            $port++;
+        }
+    }
+
+    # Test connectivity to all containers on localhost
+    for my $container (@containers) {
+        next if ($container->{name} eq "test_root_podman_6");    # XXX
+        assert_script_run "curl $curl_opts http://localhost:$container->{port}",
+          fail_message => "failed IPv$ip_version localhost test for $container->{name}";
+    }
+
+    my $iface = script_output "ip -$ip_version --json route list match default | jq -r '.[0].dev'";
+    my $ip_addr = script_output "ip -$ip_version --json addr show $iface | jq -r '.[0].addr_info[0].local'";
+    $ip_addr = "[$ip_addr]" if ($ip_version == 6);
+
+    # Test connectivity to all containers on IP address
+    for my $container (@containers) {
+        assert_script_run "curl $curl_opts http://$ip_addr:$container->{port}",
+          fail_message => "failed IPv$ip_version test for $container->{name}";
+    }
+
+    # XXX This test doesn't seem to work on o3
```

Comment:
Just checked openqaworker20 and there is no IPv6 configured (only Link-Local address).

Reply:
Fixed.